codec/common/src/cpu.cpp

```diff
@@ -309,12 +309,37 @@ uint32_t WelsCPUFeatureDetect (int32_t* pNumberOfLogicProcessors) {
 
 #elif defined(mips)
 /* for loongson */
+static uint32_t get_cpu_flags_from_cpuinfo(void)
+{
+    uint32_t flags = 0;
+
+# ifdef __linux__
+    FILE* fp = fopen("/proc/cpuinfo", "r");
+    if (!fp)
+        return flags;
+
+    char buf[200];
+    memset(buf, 0, sizeof(buf));
+    while (fgets(buf, sizeof(buf), fp)) {
+        if (!strncmp(buf, "model name", strlen("model name"))) {
+            if (strstr(buf, "3A4000")) {
+                flags |= WELS_CPU_MSA | WELS_CPU_MMI;
```

Comment:
We should probe MSA via ASEs domain.  Early Loongson-2K Kernel didn't enable MSA. If we purely rely on model, than it may lead to illegal instructions.

For all the versions of MSA enabled kernel, msa will be displayed in the ASEs Implemented.